app/configurator/components/field.tsx

```diff
@@ -254,61 +266,62 @@ export const DataFilterSelect = ({
-            <FormGroup>
-              <FormControlLabel
-                control={
-                  <MUISwitch
-                    size="small"
-                    checked={usesMostRecentValue}
-                    onChange={() =>
-                      fieldProps.onChange({
-                        target: {
-                          value: usesMostRecentValue
-                            ? `${maxValue}`
-                            : VISUALIZE_MAX_VALUE,
-                        },
-                      })
-                    }
-                  />
-                }
-                label={
-                  <Typography variant="caption">
-                    <Trans id="controls.filter.use-most-recent">
-                      Use most recent
-                    </Trans>
-                  </Typography>
-                }
-                sx={{ mr: 0 }}
-              />
-            </FormGroup>
-          </Flex>
-        ) : (
+    <div>
+      {canUseMostRecentValue ? (
+        <Flex
+          sx={{
+            width: "100%",
+            justifyContent: "space-between",
+            alignItems: "center",
+          }}
+        >
           <FieldLabel label={label} isOptional={isOptional} />
-        )
-      }
-      disabled={disabled || usesMostRecentValue}
-      options={allValues}
-      sortOptions={false}
-      controls={controls}
-      open={isOpen}
-      onClose={handleClose}
-      onOpen={handleOpen}
-      loading={loading}
-      {...fieldProps}
-      value={usesMostRecentValue ? maxValue : fieldProps.value}
-    />
+          <FormGroup>
+            <FormControlLabel
+              control={
+                <MUISwitch
+                  size="small"
+                  checked={usesMostRecentValue}
+                  onChange={() =>
+                    fieldProps.onChange({
+                      target: {
+                        value: usesMostRecentValue
+                          ? `${maxValue}`
+                          : VISUALIZE_MAX_VALUE,
+                      },
+                    })
+                  }
+                />
+              }
+              label={
+                <Typography variant="caption">
+                  <Trans id="controls.filter.use-most-recent">
+                    Use most recent
+                  </Trans>
+                </Typography>
+              }
+              sx={{ mr: 0 }}
+            />
+          </FormGroup>
+        </Flex>
+      ) : (
+        <FieldLabel label={label} isOptional={isOptional} />
+      )}
+      <Select
+        id={id}
+        disabled={disabled || usesMostRecentValue}
+        options={allValues}
+        sortOptions={false}
+        topControls={topControls}
+        sideControls={sideControls}
+        open={isOpen}
+        onClose={handleClose}
+        onOpen={handleOpen}
+        loading={loading}
+        {...fieldProps}
+        value={usesMostRecentValue ? maxValue : fieldProps.value}
+      />
+    </div>
```

Comment:
A bit difficult for me to check the diff because of the ident changes, I am wondering if we should have chart-configurator field examples in the storybook ? I think it can be used to check the "extreme" cases, for example with the screenshot I posted where a long label makes the drag handle disapppear.

Reply:
Yes, I think it would be handled in #1439 👍